Allow SwiftLint arguments, document `[linter]_command_prefix`

EfficientSetting opened the pull request:
I couldn't find a proper reason why Swiftlint arguments were "disabled" so I want ahead and implemented Swiftlint arguments so I could pipe paths in as needed.

Further, the Swiftformat command added `"."` to the end of the command no matter what, that seemed unnecessary since Swiftformat is already linting the work directory.
The Swiftformat change should have been an additional PR, but given the change is so minimal ...

samuelmeuli commented:
Thanks for the PR! The SwiftLint changes look good to me.

> Further, the Swiftformat command added "." to the end of the command no matter what, that seemed unnecessary since Swiftformat is already linting the work directory.

I don't think this is the case, see the failing test.

EfficientSetting commented:
I'll get back to you about this tomorrow. I see the test failing. Thanks

EfficientSetting commented:
@samuelmeuli I fixed the failing test, but I'm rather unsure about the quality of my fix.

The issue I'm having is that adding "." to the end of the swiftformat command, renders the command prefix option useless as the whole directory is being listed anyway. Adding the dir parameter to the end of the command, gives at least the option to input an empty string for the directory.

While I regret not having separated concerns and made multiple PRs, I also added the command_prefix docu to the Readme that I was missing while setting up lint-actions for my repo.

Jeehut commented:
Thank you @EfficientSetting for the fix, I was just running across this same problem. But I don't understand why you put two different things into one PR, so I just created https://github.com/samuelmeuli/lint-action/pull/44 so we can get the already approved SwiftLint portion merged fast.

EfficientSetting commented:
@Jeehut Like I said, it was a mistake to put three things into one PR. 
But since I fixed everything on this PR last week and made sure all tests run, I don’t see the need or reason for another PR with my changes in it. 

I’d rather have this PR merged or get a reply on what else is missing from @samuelmeuli

samuelmeuli commented on src/linters/swift-format-lockwood.js:
```diff
@@ -39,8 +39,7 @@ class SwiftFormatLockwood {
 		const fixArg = fix ? "" : "--lint";
-		return run(`${prefix} swiftformat ${fixArg} ${args} "."`, {
-			dir,
+		return run(`${prefix} swiftformat ${fixArg} ${args} ${dir}`, {
```
Documentation for `[linter]_dir`: "Directory where the linting command should be run". This would change that behavior. I think the command should be left the way it is.

EfficientSetting replied:
Fixed back to the original format

samuelmeuli commented on README.md:
```diff
@@ -148,6 +148,7 @@ jobs:
 - **`[linter]_args`**: Additional arguments to pass to the linter. Example: `eslint_args: "--max-warnings 0"` if ESLint checks should fail even if there are no errors and only warnings. Default: `""`
 - **`[linter]_dir`**: Directory where the linting command should be run. Example: `eslint_dir: server/` if ESLint is installed in the `server` subdirectory. Default: Repository root
 - **`[linter]_extensions`:** Extensions of files to check with the linter. Example: `eslint_extensions: js,ts` to lint JavaScript and TypeScript files with ESLint. Default: Varies by linter, see [`action.yml`](./action.yml)
+- **`[linter]_command_prefix`:** Command prefix to be run before the linter command. Example: `eslint_command_prefix: cat changed_files.txt |` to pipe in a list of changed files. Default: none.
```
Also, let's use the same description as in `action.yml`: "Shell command to prepend to the linter command".

samuelmeuli commented:
Thanks, @EfficientSetting! :)
